tests/stdlib/tmath.nim

```diff
@@ -350,5 +348,13 @@ template main =
     doAssert copySign(-NaN, 0.0).isNaN
     doAssert copySign(-NaN, -0.0).isNaN
 
+    when nimvm:
+      discard
+    else:
+      when not defined(js):
+        doAssert copySign(-1.0, -NaN) == 1.0
+        doAssert copySign(10.0, -NaN) == 10.0
+        doAssert copySign(1.0, copySign(NaN, -1.0)) == -1.0 # fails in VM
```

Comment:
this test actually works in VM so long nim is compiled after this PR.
Simplest is to defined `nimHasCopySign` in condsyms.nim and then:
```nim
    template fn() = doAssert copySign(1.0, copySign(NaN, -1.0)) == -1.0
    when defined(js): discard # xxx bug
    else:
      when defined(nimHasCopySign): fn()
      else:
        when nimvm: discard
        else: fn()
```

Reply:
I would rather make a follow-up PR to enable this test.

Reply:
And `tmath.nim` should support JS, I could try to enable JS tests(at least for copySign) in the net PR.

Reply:
ya we definitely need this (in another PR as you say); refs https://github.com/timotheecour/Nim/issues/494#issuecomment-751871623 ; it just requires a few disabling's for js to work (and vm already works)